### PR TITLE
name filter on project list

### LINKED ISSUE #78: Search for ProjectList
We should introduce search/filter for Projects.

We need free-text-search on Project Name, drop-down list for year of Project Start Date

* Phase 1 is free-text-search
* Phase 2 is filter by year

### PR DESCRIPTION
**Pull request by TahaKhanAbdalli**
Fixes #78

### CODE REVIEW

**Comment by IanMayo** [comment]
Hmm, I'm having a play.  If I filter for `project` - I don't get any matches.

The `FakeRest` [lib](https://github.com/marmelab/FakeRest) that this data-provider sits on _does_ provide partial matches:
image

Could you investigate and see if we can fix this?  Maybe try logging the `DataProvider`?

File: src/resources/projects/ProjectList.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {
   TopToolbar
 } from 'react-admin'
 
-const filters = [<SearchInput source='name' key='name' alwaysOn />]
+const filters = [<SearchInput source='q' key='q' alwaysOn />]
```
Comment (IanMayo) [inline src/resources/projects/ProjectList.tsx:14]: Hello @TahaKhanAbdalli - would just this change have fixed the issue?

Do we need to modify the data provider?

Comment (TahaKhanAbdalli) [reply src/resources/projects/ProjectList.tsx:14]: Yes, It by default searches all the properties but in the issue filter was specified on the name field therefore I tried to add only on the name field

Comment (IanMayo) [reply src/resources/projects/ProjectList.tsx:14]: Aah yes.  _For-now_, I'm tempted to stick with the `q` wildcard search.

**Review by IanMayo — APPROVED** [review]

**Pull request merged**